torchbenchmark/models/drq/__init__.py

```diff
@@ -0,0 +1,154 @@
+        info = dict()
+        info["internal_state"] = info_state
+        info["discount"] = 1.0
+        return (self.obs, reward, done, info)
+    def seed(self, seed=None):
+        self._norm_action_space.seed(seed)
+        self._observation_space.seed(seed)
+    def reset(self):
+        return self.obs
+    @property
+    def observation_space(self):
+        return self._observation_space
+    @property
+    def action_space(self):
+        return self._norm_action_space
+
+def make_env(cfg):
+    if cfg.env == 'ball_in_cup_catch':
+        domain_name = 'ball_in_cup'
+        task_name = 'catch'
+    elif cfg.env == 'point_mass_easy':
+        domain_name = 'point_mass'
+        task_name = 'easy'
+    else:
+        domain_name = cfg.env.split('_')[0]
+        task_name = '_'.join(cfg.env.split('_')[1:])
+     # per dreamer: https://github.com/danijar/dreamer/blob/02f0210f5991c7710826ca7881f19c64a012290c/wrappers.py#L26
+    camera_id = 2 if domain_name == 'quadruped' else 0
+
+    current_dir = os.path.dirname(os.path.realpath(__file__))
+    mockobs = pkl.load(open(os.path.join(current_dir, cfg.obs_path), "rb"))
+    low = np.amin(mockobs)
+    high = np.amax(mockobs)
+    mockobs = np.random.randint(low=11, high=228, size=mockobs.shape)
+    env = MockEnv(mockobs)
+    env = FrameStack(env, k=cfg.frame_stack)
+
+    env.seed(cfg.seed)
+    assert env.action_space.low.min() >= -1
+    assert env.action_space.high.max() <= 1
+
+    return env
+
+class Model(BenchmarkModel):
+    task = REINFORCEMENT_LEARNING.OTHER_RL
+    def __init__(self, device=None, jit=False):
+        super(Model, self).__init__()
+        self.device = device
+        self.jit = jit
+        self.cfg = DRQConfig()
+        set_seed_everywhere(self.cfg.seed)
+        self.env = make_env(self.cfg)
+        obs_shape = self.env.observation_space.shape
+        action_shape = self.env.action_space.shape
+        action_range = [
+            float(self.env.action_space.low.min()),
+            float(self.env.action_space.high.max())
+        ]
+        self.agent = DRQAgent(self.cfg, self.device, obs_shape, action_shape, action_range)
+        self.replay_buffer = ReplayBuffer(self.env.observation_space.shape,
+                                          self.env.action_space.shape,
+                                          self.cfg.replay_buffer_capacity,
+                                          self.cfg.image_pad, self.device)
+        self.step = 0
+
+    def get_module(self):
+        obs = self.env.reset()
+        obs = torch.FloatTensor(obs).to(self.device)
+        obs = obs.unsqueeze(0)
+        return self.actor, (obs, )
+
+    def train(self, niter=2):
+        episode, episode_reward, episode_step, done = 0, 0, 1, True
+        for step in range(niter):
+            obs = self.env.reset()
+            done = False
+            episode_reward = 0
+            episode_step = 0
+            episode += 1
+            if step < self.cfg.num_seed_steps:
```

Comment:
It looks like by default with niters = 2, this will always take one path? Should we have a fork based on step number for benchmarking? Do we just want to benchmark one, the other, or both?

Reply:
We would like all the code path to be covered by the default niters. In the [config](https://github.com/pytorch/benchmark/pull/387/files#diff-1c5eb1e60aafbb155ddfd1a8291eab9d24200fd71ae402157b100f4647d18093R17) file we set `num_seed_steps` to be 1. Because niters = 2 by default, both paths will be executed (once when niter=0, the other is when niter=1).

Reply:
I just want to point out that making each path execute once per benchmark is a dangerous assumption.  do you know how big num_seed_steps was by default?  

if the behavior of the model is basically to do the first path N times, and then do the second path for the rest of the time, then in the limit, probably the second path is all we really need to benchmark, or at least making them 1:1 is pretty skewing in terms of faithful representation of the workload.